Tasks/Gradle/CodeAnalysis/Common/BaseTool.ts

```diff
@@ -0,0 +1,153 @@
+
+    /**
+     * Report parser that extracts the number of affected files and the number of violations from a report
+     * 
+     * @protected
+     * @abstract
+     * @param {string} xmlReport
+     * @param {string} moduleName
+     * @returns {[number, number]} a tuple of [affected_file_count, violation_count]
+     */
+    protected abstract parseXmlReport(xmlReport: string, moduleName: string): [number, number];
+
+    public configureBuild(toolRunner: ToolRunner): ToolRunner {
+        switch (this.buildOutput.buildEngine) {
+            case BuildEngine.Maven: {
+                throw new Error('Not implemented');
+            }
+            case BuildEngine.Gradle: {
+
+                if (this.isEnabled()) {
+                    console.log(tl.loc('codeAnalysis_ToolIsEnabled'), this.toolName);
+
+                    var initScriptPath: string = path.join(__dirname, '../', this.toolName.toLowerCase() + '.gradle');
+                    toolRunner.arg(['-I', initScriptPath]);
+                    break;
+                }
+            }
+        }
+        return toolRunner;
+    }
+
+
+    public processResults(): AnalysisResult[] {
+
+        if (!this.isEnabled()) {
+            tl.debug(`[CA] ${this.toolName} analysis is not enabled`);
+            return [];
+        }
+
+        var results: AnalysisResult[] = [];
+        var outputs: ModuleOutput[] = this.buildOutput.findModuleOutputs();
+        tl.debug(`[CA] ${this.toolName} parser found ${outputs.length} possible modules`);
+
+        for (var output of outputs) {
+            var result = this.parseModuleOutput(output);
+
+            if (result) {
+                results.push(result);
+            }
+        }
+
+        return results;
+    }
+
+
+    protected /* for test purposes, otherwise private */ isEnabled(): boolean {
+        return tl.getBoolInput(this.uiInputName, false);
+    }
+
+    private parseModuleOutput(output: ModuleOutput): AnalysisResult {
+
+        let reportDir = this.getBuildReportDir(output);
+        let xmlReports = glob.sync(path.join(reportDir, '*.xml'));
+
+        if (xmlReports.length === 0) {
+            tl.debug(`[CA] No ${this.toolName} reports found for the ${output.moduleName} module. Searched in ${reportDir}`);
+            return null;
+        }
+
+        tl.debug(`[CA] Found ${xmlReports.length} xml reports for module ${output.moduleName}`)
+        return this.buildAnalysisResultFromModule(xmlReports, output.moduleName);
+    }
+
+    private buildAnalysisResultFromModule(xmlReports: string[], moduleName: string): AnalysisResult {
+
+        let analysisResult: AnalysisResult = null;
+        let fileCount: number = 0;
+        let violationCount: number = 0;
+        let artifacts: string[] = [];
+
+        for (var xmlReport of xmlReports) {
+
+            var result = this.parseXmlReport(xmlReport, moduleName);
+
+            if (result && (result[0] !== 0)) {
```

Comment:
Is there any console logging to clearly say "empty report not uploaded", or is the system silent when there are no issues? If the latter happens, I would say this is a problem in the experience.